### PR TITLE
[VEN-794]Fix/governor audit

### PR DESCRIPTION
**Pull request by 0xlucian**
## Description
This PR reflects changes from Venus Governor Update PR Audit Report. The fix for each issue is packed in a separate commit.

## Checklist

- [ ] I have updated the documentation to account for the changes in the code.
- [ ] If I added new functionality, I added tests covering it.
- [ ] If I fixed a bug, I added a test preventing this bug from silently reappearing again.
- [ ] My contribution follows [Venus contribution guidelines](docs/CONTRIBUTING.md).

### CODE REVIEW

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 1]
```diff
@@ -61,8 +62,8 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
         guardian = guardian_;
 
         //Set parameters for each Governance Route
-        for(uint256 i; i<proposalConfigs_.length; ++i){
-            uint _votingPeriod = proposalConfigs_[i].votingPeriod;
+        uint256 arrLenght = proposalConfigs_.length;
+        for(uint256 i; i<arrLenght; ++i){
```
Comment (coreyar) [inline contracts/Governance/GovernorBravoDelegate.sol:66]:
```suggestion
        uint256 arrLength = proposalConfigs_.length;
        for(uint256 i; i<arrLength; ++i){
```

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 2]
```diff
@@ -78,17 +79,20 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
 
     /**
       * @notice Function used to propose a new proposal. Sender must have delegates above the proposal threshold
+      * @dev NOTE: Proposals with duplicate set of actions could not be queued for execution. If the proposals consists
```
Comment (coreyar) [inline contracts/Governance/GovernorBravoDelegate.sol:82]:
```suggestion
      * @dev NOTE: Proposals with duplicate set of actions can not be queued for execution. If the proposals consists 
```

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 3]
```diff
@@ -78,17 +79,20 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
 
     /**
       * @notice Function used to propose a new proposal. Sender must have delegates above the proposal threshold
+      * @dev NOTE: Proposals with duplicate set of actions could not be queued for execution. If the proposals consists 
+      *  of repetitive actions, its recommended to split those actions into separate proposals
```
Comment (coreyar) [inline contracts/Governance/GovernorBravoDelegate.sol:83]:
```suggestion
      *  of duplicate actions, its recommended to split those actions into separate proposals
```

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 4]
```diff
@@ -61,8 +62,8 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
         guardian = guardian_;
 
         //Set parameters for each Governance Route
-        for(uint256 i; i<proposalConfigs_.length; ++i){
-            uint _votingPeriod = proposalConfigs_[i].votingPeriod;
+        uint256 arrLength = proposalConfigs_.length;
+        for(uint256 i; i<arrLength; ++i){
```
Comment (kkirka) [inline contracts/Governance/GovernorBravoDelegate.sol:66]:
```suggestion
        for (uint256 i; i < arrLength; ++i) {
```

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 5]
```diff
@@ -78,17 +79,20 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
 
     /**
       * @notice Function used to propose a new proposal. Sender must have delegates above the proposal threshold
+      * @dev NOTE: Proposals with duplicate set of actions can not be queued for execution. If the proposals consists 
+      *  of duplicate actions, its recommended to split those actions into separate proposals
```
Comment (kkirka) [inline contracts/Governance/GovernorBravoDelegate.sol:83]:
```suggestion
      *  of duplicate actions, it's recommended to split those actions into separate proposals
```

**Review by kkirka — APPROVED** [review]

**Review by defcon022 — APPROVED** [review]

File: contracts/Governance/GovernorBravoDelegate.sol [hunk 6]
```diff
@@ -78,17 +79,20 @@ contract GovernorBravoDelegate is GovernorBravoDelegateStorageV2, GovernorBravoE
...
+      * @param proposalType the type of the proposal (e.g NORMAL, FASTTRACK, CRITICAL)
       * @return Proposal id of new proposal
       */
     function propose(address[] memory targets, uint[] memory values, string[] memory signatures, bytes[] memory calldatas, string memory description, ProposalType proposalType) public returns (uint) {
         // Reject proposals before initiating as Governor
         require(initialProposalId != 0, "GovernorBravo::propose: Governor Bravo not active");
-        require(xvsVault.getPriorVotes(msg.sender, sub256(block.number, 1)) > proposalConfigs[uint8(proposalType)].proposalThreshold, "GovernorBravo::propose: proposer votes below proposal threshold");
+        require(xvsVault.getPriorVotes(msg.sender, sub256(block.number, 1)) >= proposalConfigs[uint8(proposalType)].proposalThreshold, "GovernorBravo::propose: proposer votes below proposal threshold");
         require(targets.length == values.length && targets.length == signatures.length && targets.length == calldatas.length, "GovernorBravo::propose: proposal function information arity mismatch");
         require(targets.length != 0, "GovernorBravo::propose: must provide actions");
```
Comment (defcon022) [inline contracts/Governance/GovernorBravoDelegate.sol:97]: `target.length > 0` check takes less gas then `target.length != 0`

**Pull request merged**